### DISCUSSION TITLE
training error

### DISCUSSION THREAD

**Discussion by m11113040** [open-edge-platform/anomalib#1802] [Q&A] [answered]
Can someone help me with the training steps for custom data? I keep encountering an error 

I DO these:
# Use of virtual environment is highy recommended
# Using conda
yes | conda create -n anomalib_env python=3.10
conda activate anomalib_env
# Clone the repository and install in editable mode
git clone https://github.com/openvinotoolkit/anomalib.git
cd anomalib
pip install -e .
# Install the full package
anomalib install

than,go to (https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/data/custom_data.html) set my own data
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/6d9fed95-b6a8-4776-8d6f-485f3b8d4716)

after that i run (anomalib train --data configs/normal.yaml --model anomalib.models.Padim --task CLASSIFICATION)
it always error in class_path
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/cb697ae1-fb3b-45b3-a8e4-e187631231ad)

and my normal pictures is in C:/Users/CygerMan/anomalib/data/good

**Comment by abc-125** [comment]
Hello, I think you need to add the key `name` to your config, like this:
```
class_path: anomalib.data.Folder
init_args:
  name: bottle
  root: "datasets/MVTec/bottle"
```

**Reply by samet-akcay** [reply]
Yeah, we have recently added `name` to `Folder` in PR https://github.com/openvinotoolkit/anomalib/pull/1773 to properly save the training results.

This will hopefully be the last breaking change.

**Reply by m11113040** [reply]
Thanks for your help,I renew my config,but i has a normalization problem,and 
 (https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/data/image/folder.html)  didnt explain about normalization 

> class_path: anomalib.data.Folder
> init_args:
>   name: metal
>   root: "C:/Users/CygerMan/anomalib/m11113040/metal"
>   normal_dir: "train/good"
>   normalization: imagenet
>   test_split_mode: SYNTHETIC
>   task: CLASSIFICATION

this is an example:
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/a9d540ae-ba37-4e0f-b6da-fae48fcffa15)

run error:
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/67ea9792-38e1-4d8e-8922-fb352fda40c4)

**Reply by blaz-r** [reply]
It seems like normalization is no longer a part of dataset class but rather part of each model. This is reflected in docs but not in config.

**Reply by m11113040** [reply]
so,what should i do to fix this problem

**Reply by blaz-r** [reply]
I think you can just remove the normalization from config, it's part of the model and doesn't need to be configured.

**Reply by m11113040** [reply]
thanks sir,when i stop on this step ,it seems to show my imwrite problem?
so,i have to check my run.py or ....?
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/fb97aaf1-6023-4e92-a793-54ad66ad6e17)
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/7c86509b-81ed-4803-969a-95fd4d2095f3)

**Reply by blaz-r** [reply]
Plese share the entire error message.

**Reply by m11113040** [reply]
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/3234239d-a9ba-46c8-8ce9-ba89048892b2)
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/fb9705d2-099d-4008-845d-c8cff61a304f)
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/4b4d01df-7051-41b5-8b6b-7d573fa440d4)
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/659587a9-b5b3-49f5-aaa6-0a2043ae19cc)
![image](https://github.com/openvinotoolkit/anomalib/assets/157609224/2266db7d-622d-4596-a244-c23e0aa1d28d)

**Reply by blaz-r** [reply]
As the error suggest, there is a problem with file name. I'm not sure why that happens, we'll need to investigate.

**Comment by Phyonna** [comment]
hello ,I meet this problem,too. Have you solved the problem of imwrite?

**Comment by Phyonna** [comment, marked as answer]
I have solved.The problem file is : anomalib\src\anomalib\data\utils\image.py.
In this file , the path of image is wrong.I changed it in to this , hope it can help you.
<img width="422" alt="a68e9704401b4db5e6927d161fc38d0" src="https://github.com/openvinotoolkit/anomalib/assets/102140144/beb21d48-c698-4e5b-9708-57edc9c83a39">

**Reply by m11113040** [reply]
thanks,i will try it

**Reply by zhaozzt** [reply]
![5](https://github.com/openvinotoolkit/anomalib/assets/127210381/0a55d05c-9b40-4911-847e-c5f30288486a)
I encountered such an error, please help me, thank you

**Reply by Phyonna** [reply]
sorry to hear that. I remenber the error , but can‘t remenber how I solved it. 
By the way , I met a lot of errors with the V1.0.0 , so I try V0.7.0 , and that works very well . If you can't solve the errors , I think maybe you can have a try with the V0.7.0 .

**Reply by blaz-r** [reply]
Hi,
since the error seems to be in shape of the operands, I recommend you go through with debugger and check where this discrepancy occurs to potentially see why. Maybe compare 0.7 shapes to 1.0. If you find the issue, please share it here, or even better open a PR with fix. Thanks a lot 😄
